app/controllers/spotlight/resources/csv_upload_controller.rb

```diff
@@ -14,7 +14,7 @@ class CsvUploadController < ApplicationController
 
       def create
         file = csv_params[:url]
-        csv = CSV.parse(file.read, headers: true, return_headers: false, encoding: 'utf-8').map(&:to_hash)
+        csv = CSV.parse(File.read(file.tempfile.to_path.to_s, encoding: 'UTF-8'), headers: true, return_headers: false, encoding: 'utf-8').map(&:to_hash)
```

Comment:
Are we sure all uploads have a backing `tempfile`? The Rails documentation suggests in some cases it'll be a simple `StringIO` object: http://guides.rubyonrails.org/form_helpers.html#uploading-files